### PR TITLE
Fix layouts for mobile screens

### PR DESCRIPTION
**Pull request by JustinBeBoy**
This PR is a continuation of #963 

This PR fixs: 

- [x] When the software keyboard is showing, the hide balance icon is hidden
- [x] The "transaction" text on the bottom nav is not showing fully
- [x] When changing setings(dark mode, currency conversion, language, etc...) the bottom nav gets distorted
- [x] Fix not show all seeds word option in verify seed words page
- [x] Make 2 columns of 33 seeds word in save seed page on mobile
- [x] Fix dark mode not working when change from setting
- [x] Add margin to title and buttons in start page
- [x] Fix button "View on politeia" in proposal detail page not show
- [x] Fix "Clear all" and "validate wallet seeds" button not show when keyboard appears in restore wallet page
- [x] Close the gap to increase the visibility of information in many page
- [x] Right margin and items of ticket record are uneven in staking page

**Screenshots**

| | |
|-|-|

| | |
|-|-|

| | |
|-|-|

| | |
|-|-|

| | |
|-|-|

| | |
|-|-|

### CODE REVIEW

File: ui/page/seedbackup/save_seed.go [hunk 1]
```diff
@@ -90,6 +91,22 @@ func (pg *SaveSeedPage) OnNavigatedTo() {
 				row2 := wordList[11:22]
 				row3 := wordList[22:]
 
+				//for mobile
+				rowm1 := wordList[:17]
+				rowm2 := wordList[17:]
```
Comment (dreacot) [inline ui/page/seedbackup/save_seed.go]:
```suggestion
				rowMobile1 := wordList[:17]
				rowMobile2 := wordList[17:]
```

File: ui/values/colors.go [hunk 2]
```diff
@@ -71,54 +71,56 @@ func (c *Color) DarkThemeColors() {
...
 
-func (c *Color) DefualtThemeColors() *Color {
-	c.Primary = rgb(0x2970ff)
-	c.Primary50 = rgb(0xE3F2FF)
-	c.PrimaryHighlight = rgb(0x1B41B3)
+func (c *Color) DefaultThemeColors() *Color {
+	cl := Color{
+		Primary:          rgb(0x2970ff),
+		Primary50:        rgb(0xE3F2FF),
+		PrimaryHighlight: rgb(0x1B41B3),
```
Comment (dreacot) [inline ui/values/colors.go:78]: why this change, what aren't we using `c`

**Review by dreacot — CHANGES_REQUESTED** [review]
Left some comments, other than that, this looks good

File: ui/values/colors.go (hunk 2 above)
Comment (JustinBeBoy) [reply ui/values/colors.go:78]: DefaultThemeColors function should to get default color, it shouldn't a set color function because some page use this function for get default color

File: ui/page/seedbackup/verify_seed.go [hunk 3]
```diff
@@ -40,7 +40,7 @@ type VerifySeedPage struct {
 
 	backButton   decredmaterial.IconButton
 	actionButton decredmaterial.Button
-	seedList     *layout.List
+	listSeedList []*layout.List
```
Comment (dreacot) [inline ui/page/seedbackup/verify_seed.go]: this is not directly clear to me what this does, we could use a ore appropriate name, or add a comment

**Review by dreacot — CHANGES_REQUESTED** [review]
Overall this looks great. some minor touch ups and i'd merge

1. The button at the bottom right doesn't have the correct bottom and right margins


2. The floating send and receive button, doesn't have the correct the correct bottom margin


3. This page doesn't have the proper left and right margins


4. this item can be made full width

File: ui/decredmaterial/theme.go [hunk 4]
```diff
@@ -74,9 +74,8 @@ func NewTheme(fontCollection []text.FontFace, decredIcons map[string]image.Image
 }
 
 func (t *Theme) SwitchDarkMode(isDarkModeOn bool, decredIcons map[string]image.Image) {
-	t.Color.DefualtThemeColors()
+	t.Color = t.Color.DefaultThemeColors()
```
Comment (Sirmorrison) [inline ui/decredmaterial/theme.go:77]: I dont see the reason for this change. if it does not affect your code in anyway, kindly revert.

File: ui/decredmaterial/progressbar.go [hunk 5]
```diff
@@ -152,7 +152,10 @@ func (mp *MultiLayerProgressBar) progressBarLayout(gtx C) D {
 	}
 
 	calProgressWidth := func(progress int) int {
-		val := (progress / mp.total) * 100
+		val := 1
+		if mp.total != 0 {
+			val = (progress / mp.total) * 100
+		}
```
Comment (Sirmorrison) [inline ui/decredmaterial/progressbar.go:158]: Why are we making this update?

File: ui/page/components/components.go [hunk 6]
```diff
@@ -108,6 +108,31 @@ func UniformHorizontalPadding(gtx layout.Context, body layout.Widget) layout.Dim
...
+	if withList {
+		padding = 0
+	}
+
+	return layout.Inset{
+		Top:   values.MarginPadding24,
+		Right: padding,
+		Left:  values.MarginPadding10,
+	}.Layout(gtx, body)
+}
```
Comment (Sirmorrison) [inline ui/page/components/components.go]: The only difference i see with the 2 methods is the top inset. this can be merged into 1 and the inset passed as a parameter.

```suggestion
func UniformMobile(gtx layout.Context, isHorizontal, withList bool, body layout.Widget) layout.Dimensions {

	insetRight := values.MarginPadding10
	if withList {
		insetRight = values.MarginPadding0
	}
	
	insetTop := values.MarginPadding24
	if isHorizontal{
	     insetTop = values.MarginPadding0
	}
	
	return layout.Inset{
		Top:   insetTop,
		Right: insetRight,
		Left:  values.MarginPadding10,
	}.Layout(gtx, body)
}
```

File: ui/page/debug_page.go [hunk 7]
```diff
@@ -162,7 +162,18 @@ func (pg *DebugPage) Layout(gtx C) D {
...
+	}
+	return pg.layoutDesktop(gtx, container)
+}
+
+func (pg *DebugPage) layoutDesktop(gtx layout.Context, body layout.Widget) layout.Dimensions {
+	return components.UniformPadding(gtx, body)
+}
+
+func (pg *DebugPage) layoutMobile(gtx layout.Context, body layout.Widget) layout.Dimensions {
+	return components.UniformMobile(gtx, false, body)
```
Comment (Sirmorrison) [inline ui/page/debug_page.go]: I dont see the need for the `layoutDesktop` and `layoutMobile` functions as they are only used once.

```suggestion
	if pg.Load.GetCurrentAppWidth() <= gtx.Dp(values.StartMobileView) {
               // mobile layout
		return components.UniformMobile(gtx, false, container)
	}
	// desktop layout
	return components.UniformPadding(gtx, container) 
}
	
```

File: ui/page/governance/consensus_page.go [hunk 8]
```diff
@@ -275,6 +282,73 @@ func (pg *ConsensusPage) Layout(gtx C) D {
...
+						return pg.orderDropDown.Layout(gtx, 55, true)
+					}),
+					layout.Expanded(func(gtx C) D {
+						return pg.walletDropDown.Layout(gtx, pg.orderDropDown.Width+51, true)
+					}),
+				)
+			})
+		}),
+	)
+}
```
Comment (Sirmorrison) [inline ui/page/governance/consensus_page.go:350]: I dont see the difference between what you have here and on the desktop layout.. Kindly help clarify why we need this?

Actually lookign again i see the dropdown position values changed if that be the only case, we can have only on function that takes the dropdown position as a parameter and the values can be passed to it by the different layouts.

```
 func (pg *ConsensusPage) pageLayout(gtx layout.Context, orderPos int, walletPos int) layout.Dimensions {}
```

```
if pg.Load.GetCurrentAppWidth() <= gtx.Dp(values.StartMobileView) {
		return pg.pageLayout(gtx, val, val)
	}
   return pg.pageLayout(gtx, val, val)
```

File: ui/page/governance/governance_page.go [hunk 9]
```diff
@@ -117,6 +124,24 @@ func (pg *Page) Layout(gtx C) D {
...
+			layout.Rigid(pg.layoutTabs),
+			layout.Rigid(pg.Theme.Separator().Layout),
+			layout.Flexed(1, func(gtx C) D {
+				return layout.Inset{Top: values.MarginPadding16}.Layout(gtx, func(gtx C) D {
+					return pg.CurrentPage().Layout(gtx)
+				})
+			}),
+		)
+	})
+}
```
Comment (Sirmorrison) [inline ui/page/governance/governance_page.go]: No need for this as i see the only use was is in the splash screen `uniformMobileLayout()`

File: ui/page/governance/governance_page.go [hunk 10]
```diff
@@ -99,6 +99,13 @@ func (pg *Page) HandleUserInteractions() {
...
+	if pg.Load.GetCurrentAppWidth() <= gtx.Dp(values.StartMobileView) {
+		return pg.layoutMobile(gtx)
+	}
+	return pg.layoutDesktop(gtx)
+}
+
+func (pg *Page) layoutDesktop(gtx layout.Context) layout.Dimensions {
 	if !pg.isGovernanceFeatureEnabled() {
 		return components.UniformPadding(gtx, pg.splashScreenLayout)
 	}
```
Comment (Sirmorrison) [inline ui/page/governance/governance_page.go:111]:
```suggestion

if !pg.isGovernanceFeatureEnabled() {
	if pg.Load.GetCurrentAppWidth() <= gtx.Dp(values.StartMobileView) {
	    return components.UniformMobile(gtx, false, pg.splashScreenLayout)
	}
		return components.UniformPadding(gtx, pg.splashScreenLayout)
}
```

File: ui/page/governance/proposal_details_page.go [hunk 11]
```diff
@@ -579,3 +586,59 @@ func (pg *ProposalDetails) Layout(gtx C) D {
...
+			if proposal.IndexFile != "" && proposal.IndexFileVersion == proposal.Version {
+				proposalDescription = proposal.IndexFile
+			} else {
+				var err error
+				proposalDescription, err = pg.WL.MultiWallet.Politeia.FetchProposalDescription(proposal.Token)
+				if err != nil {
+					log.Errorf("Error loading proposal description: %v", err)
+					time.Sleep(7 * time.Second)
+					pg.loadingDescription = false
+					return
```
Comment (Sirmorrison) [inline ui/page/governance/proposal_details_page.go:606]: Same review above applies here.

File: ui/page/governance/proposals_page.go [hunk 12]
```diff
@@ -264,6 +271,57 @@ func (pg *ProposalsPage) Layout(gtx C) D {
...
+					// 	})
+					// }),
+					layout.Expanded(func(gtx C) D {
+						gtx.Constraints.Min.X = gtx.Constraints.Max.X
+						return layout.E.Layout(gtx, func(gtx C) D {
+							card := pg.Theme.Card()
+							card.Radius = decredmaterial.Radius(8)
+							return layout.Inset{Right: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
+								return card.Layout(gtx, func(gtx C) D {
+									return layout.UniformInset(values.MarginPadding8).Layout(gtx, pg.layoutSyncSection)
```
Comment (Sirmorrison) [inline ui/page/governance/proposals_page.go:308]: These mobile layouts are just a bunch or repeated codes. Lets have them optimized as mentioned above

**Review by Sirmorrison — DISMISSED** [review]
These mobile layouts are just a bunch or repeated codes with little to not significant updates. Lets have them optimized as mentioned in the comments.

File: ui/page/governance/proposals_page.go (hunk 12 above)
Comment (dreacot) [reply ui/page/governance/proposals_page.go:308]: For now they might look like they are, but they are not. currently we are trying to share the same layouts for desktop and mobile, and making some optimizations to the code in order to make it look somewhat presentable on mobile. However, we are going to be implementing totally different UIs for mobile further down the line when proper mockups become available. The mobile and desktop views need to be clearly distinguishable.

This PR, the ones before it, and the ones coming after it, are aimed at setting the bedrock foundation for the godcr mobile builds

Having said that the current layout separation is in line with the scope of this PR. We should leave as is.

File: ui/page/governance/proposal_details_page.go (hunk 11 above)
Comment (dreacot) [reply ui/page/governance/proposal_details_page.go:606]: we should leave as is, further details here https://github.com/planetdecred/godcr/pull/971#discussion_r904149266

File: ui/page/governance/governance_page.go (hunk 10 above)
Comment (dreacot) [reply ui/page/governance/governance_page.go:111]: we should leave as is, further details here https://github.com/planetdecred/godcr/pull/971#discussion_r904149266

File: ui/page/governance/governance_page.go (hunk 9 above)
Comment (dreacot) [reply ui/page/governance/governance_page.go]: we should leave as is, further details here https://github.com/planetdecred/godcr/pull/971#discussion_r904149266

File: ui/page/governance/consensus_page.go (hunk 8 above)
Comment (dreacot) [reply ui/page/governance/consensus_page.go:350]: we should leave as is, further details here https://github.com/planetdecred/godcr/pull/971#discussion_r904149266

File: ui/page/debug_page.go (hunk 7 above)
Comment (dreacot) [reply ui/page/debug_page.go]: we should leave as is, further details here https://github.com/planetdecred/godcr/pull/971#discussion_r904149266

File: ui/page/seedbackup/save_seed.go (hunk 1 above)
Comment (JustinBeBoy) [reply ui/page/seedbackup/save_seed.go]: I have change DefualtThemeColors() function with its true purpose, I will return new object of color, it shouldn't change current color because some function in dark mode use it to get the default color

File: ui/decredmaterial/progressbar.go (hunk 5 above)
Comment (JustinBeBoy) [reply ui/decredmaterial/progressbar.go:158]: in my case for some reason mp.total = 0 causes the app to crash when switching to staking tab, it should be checked before doing the division

File: ui/decredmaterial/theme.go (hunk 4 above)
Comment (JustinBeBoy) [reply ui/decredmaterial/theme.go:77]: The DefualtThemeColors function should return object of default colors, it shouldn't change current colors of theme because some page in app use this function to get default color

Comment (dreacot) [reply ui/decredmaterial/theme.go:77]: @JustinBeBoy perhaps you can create a new issue for this, and have it solved in a separate PR, properly documenting the reason for the change. As the added change does not affect this PR/ is not relating to this PR, it's outside the scope of this PR.

File: ui/page/components/components.go [hunk 13]
```diff
@@ -108,27 +108,31 @@ func UniformHorizontalPadding(gtx layout.Context, body layout.Widget) layout.Dim
...
+// func UniformMobileHorizontalPadding(gtx layout.Context, withList bool, body layout.Widget) layout.Dimensions {
+// 	padding := values.MarginPadding10
+// 	if withList {
+// 		padding = 0
+// 	}
+// 	return layout.Inset{
+// 		Right: padding,
+// 		Left:  values.MarginPadding10,
+// 	}.Layout(gtx, body)
+// }
```
Comment (dreacot) [inline ui/page/components/components.go]: let's remove this comment if it's not needed

File: ui/decredmaterial/theme.go (hunk 4 above)
Comment (JustinBeBoy) [reply ui/decredmaterial/theme.go:77]: It's not outside the scope of this PR because it is the cause of problems with dark mode

Comment (dreacot) [reply ui/decredmaterial/theme.go:77]: oh I see, thanks for pointing that out, i thought it was just some code optimization

File: ui/page/debug_page.go (hunk 7 above)
Comment (Sirmorrison) [reply ui/page/debug_page.go]: Then i believe it wont be out of order to properly comments these codes for easier reviews.

File: ui/page/governance/consensus_page.go (hunk 8 above)
Comment (Sirmorrison) [reply ui/page/governance/consensus_page.go:350]: Then i believe it wont be out of order to properly comments these codes for easier reviews

File: ui/page/governance/proposals_page.go (hunk 12 above)
Comment (Sirmorrison) [reply ui/page/governance/proposals_page.go:308]: Then i believe it wont be out of order to properly comments these codes for easier reviews.

**Review by dreacot — CHANGES_REQUESTED** [review]
align the text and icon left, put them in same row


just like we have here

File: ui/decredmaterial/progressbar.go (hunk 5 above)
Comment (dreacot) [reply ui/decredmaterial/progressbar.go:158]: i think converting to float should fix the crash

File: ui/page/debug_page.go (hunk 7 above)
Comment (dreacot) [reply ui/page/debug_page.go]: yeah, your right

**Review by dreacot — APPROVED** [review]
LGTM
